### PR TITLE
Rename -mabi=sandbox to -mabi=purecap

### PR DESCRIPTION
**Pull request by kgudka**

### CODE REVIEW

File: include/clang/Driver/Options.td [hunk 1]
```diff
@@ -2024,6 +2024,9 @@ def mglibc : Flag<["-"], "mglibc">, Group<m_libc_Group>, Flags<[HelpHidden]>;
 def muclibc : Flag<["-"], "muclibc">, Group<m_libc_Group>, Flags<[HelpHidden]>;
 def module_file_info : Flag<["-"], "module-file-info">, Flags<[DriverOption,CC1Option]>, Group<Action_Group>,
   HelpText<"Provide information about a particular module file">;
+def msandbox : Flag<["-"], "msandbox">, Group<m_Group>,
+  HelpText<"CHERI Pure Capability ABI generated code will be run in a sandbox (CHERI only)">,
```
Comment (davidchisnall) [inline include/clang/Driver/Options.td]: `-msandbox` shouldn't imply `-mabi=purecap` - I think n64 code is currently broken in sandboxes, but we should support hybrid ABI code in sandboxes again.

File: lib/Driver/ToolChains/FreeBSD.cpp [hunk 2]
```diff
@@ -164,7 +164,7 @@ void freebsd::Linker::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-Bshareable");
     } else {
       CmdArgs.push_back("-dynamic-linker");
-      if (IsSandboxABI)
+      if (IsCheriPureCapABI)
         CmdArgs.push_back("/libexec/ld-cheri-elf.so.1");
```
Comment (davidchisnall) [inline lib/Driver/ToolChains/FreeBSD.cpp]: We probably don't want anything in this block for -msandbox.

File: lib/Driver/ToolChains/FreeBSD.cpp [hunk 3]
```diff
@@ -204,7 +204,7 @@ void freebsd::Linker::ConstructJob(Compilation &C, const JobAction &JA,
 
   // The FreeBSD/MIPS version of GNU ld is horribly buggy and errors out
   // complaining about linking 32-bit and 64-bit code when linking CHERI code.
-  if (IsSandboxABI)
+  if (IsCheriPureCapABI)
     CmdArgs.push_back(Args.MakeArgString("--no-warn-mismatch"));
```
Comment (davidchisnall) [inline lib/Driver/ToolChains/FreeBSD.cpp]: I wonder if we still need this with lld?

File: include/clang/Driver/Options.td (hunk 1 above)
Comment (kgudka) [inline include/clang/Driver/Options.td]: I have fixed this in 86f9eaa. Should we also temporarily output a compiler error if someone specifies -msandbox without -mabi=purecap?

**Comment by kgudka** [comment]
I've just committed a test in f7323b5

**Comment by davidchisnall** [comment]
Looks good to me.

File: lib/Driver/ToolChains/FreeBSD.cpp (hunk 3 above)
Comment (RichardsonAlex) [inline lib/Driver/ToolChains/FreeBSD.cpp]: I think this might still be needed as long as we build most of our MIPS binaries with GCC

**Pull request merged**